avocado/utils/genio.py

```diff
@@ -174,3 +174,22 @@ def write_one_line(filename, line):
     :type line: str
     """
     write_file(filename, line.rstrip('\n') + '\n')
+
+
+def safe_write_file(filename, data):
+    """
+    Write data to a file.
```

Comment:
Btw are you sure you want to just log the failure instead of failing the test? I mean when I fail to write my value it probably means the test won't use correct setting...